### PR TITLE
targetSdk 31 part 2

### PR DESCRIPTION
**Pull request by bekuno**
Set targetSdkVersion 31 also for the other sub projects and the tests. 
Additionally, set minSdkVersion 21 for tests and fix typo.

### CODE REVIEW

File: main/build.gradle [hunk 1]
```diff
@@ -29,8 +29,8 @@ android {
 
     defaultConfig {
         minSdkVersion 21
-        targetSdk 31
```
Comment (moving-bits) [inline main/build.gradle:32]: FYI: That's what Android Studio changed `build.gradle` to when I changed the targetSdk in project settings. 
Any difference between the two?

Comment (bekuno) [reply main/build.gradle:32]: > FYI: That's what Android Studio changed `build.gradle` to when I changed the targetSdk in project settings. Any difference between the two?

Interesting, I did not got this change.
What I found is https://stackoverflow.com/questions/67250362/what-is-difference-between-compilesdk-and-compilesdkversion-in-android-studio-gr/67251145#67251145 .
But in the release notes for AGP is nothing written for it.

I tested also the update to AGP 7.* (in small steps and in one step) with the upgrade assistent, but no change in this was was suggested/made.
If we really want to use the short names, then we need first the update to AGP 7 as written in the above stackoverflow text, and then we should change all 3 *Version names.

Comment (moving-bits) [reply main/build.gradle:32]: That's weird, as I have not upgraded AGP (and I'm also still running AS 2021.1.1) - don't know why it has inserted the short form.

But anyway, I do not see any need to use the short form currently, so let's stay with the long form for now. Thanks for fixing it.

Comment (bekuno) [reply main/build.gradle:32]: > FYI: That's what Android Studio changed `build.gradle` to when I changed the targetSdk in project settings.
> Any difference between the two?

I could reproduce it in AS for all tree Versions names with the "Project Structure" dialog.

**Pull request merged**